pystac_client/client.py

```diff
@@ -450,8 +450,9 @@ def get_items(
 
         Args:
             ids: Zero or more item ids to find.
-            recursive: unused in pystac-client, but needed for falling back to pystac
-
+            recursive: If this client conforms to the ITEM_SEARCH conformance class,
+                this is unused and this will always yield items recursively.
+                Otherwise, this will only return items recursively if True or None.
```

Comment:
Sorry to bikeshed this PR, but can we raise a warning if `recursive = Ralse` but we use `ITEM_SEARCH`, aka put this docstring into code via a warning?